hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -498,3 +505,39 @@ func defaultWorkerSecurityGroupRules(machineCIDRs []string) []capo.SecurityGroup
 
 	return ingressRules
 }
+
+// lookupRHCOSImage looks up a release image and extracts the RHCOS image URL and release version
+func lookupRHCOSImage(ctx context.Context, client client.Client, hcluster *hyperv1.HostedCluster) (string, string, error) {
```

Comment:
We should just leverage the existing `lookupRHCOSImage` and make it an exported function. We could pass in a platform field to differentiate between OSP and Azure.